### PR TITLE
add publishing ability to auth stack

### PR DESCRIPTION
**Pull request by dionisislolas**
Add publishing ability to auth stack

### CODE REVIEW

File: v2/stacks/auth/.env [hunk 1]
```diff
@@ -14,7 +14,8 @@ PATH_MANIFESTS="../../manifests"
 PATH_PROVISIONING="../../provisioning"
 
 # -- Global variable overrides
-IS_PUBLISHING="Y"
+IS_PUBLISHING=true
```
Comment (lindenmckenzie) [inline v2/stacks/auth/.env:17]: I'm fairly sure Babbage uses Y/N to do feature flags, so why are you changing this?

File: v2/stacks/auth/core-ons.yml [hunk 2]
```diff
@@ -10,9 +10,93 @@ services:
...
-      ENABLE_NEW_SIGN_IN:     "true"
+      ENABLE_NEW_SIGN_IN: "true"
+  dp-frontend-router:
+    extends:
+      file: ${PATH_MANIFESTS}/core-ons/dp-frontend-router.yml
+      service: dp-frontend-router
+  dp-frontend-homepage-controller:
+    extends:
+      file: ${PATH_MANIFESTS}/core-ons/dp-frontend-homepage-controller.yml
+      service: dp-frontend-homepage-controller
```
Comment (lindenmckenzie) [inline v2/stacks/auth/core-ons.yml:26]: Do you need the homepage-controller to do publishing?

File: v2/stacks/auth/core-ons.yml [hunk 3]
```diff
@@ -10,9 +10,93 @@ services:
...
+    depends_on:
+      - zookeeper-1
+  dp-design-system:
+    extends:
+      file: ${PATH_MANIFESTS}/core-ons/dp-design-system.yml
+      service: dp-design-system
+  dp-topic-api:
+    extends:
+      file: ${PATH_MANIFESTS}/core-ons/dp-topic-api.yml
+      service: dp-topic-api
```
Comment (lindenmckenzie) [inline v2/stacks/auth/core-ons.yml:80]: Do you need the dp-topic-api to do publishing?

File: v2/stacks/auth/core-ons.yml [hunk 4]
```diff
@@ -10,9 +10,93 @@ services:
...
+      file: ${PATH_MANIFESTS}/deps/kafka-confluentinc.yml
+      service: kafka-2
+    depends_on:
+      - zookeeper-1
+  kafka-3:
+    extends:
+      file: ${PATH_MANIFESTS}/deps/kafka-confluentinc.yml
+      service: kafka-3
+    depends_on:
+      - zookeeper-1
```
Comment (lindenmckenzie) [inline v2/stacks/auth/core-ons.yml:72]: Supporting services (non-apps) should be in the `deps.yml` file - some others lower down as well.

File: v2/stacks/auth/core-ons.yml [hunk 5]
```diff
@@ -10,9 +10,93 @@ services:
...
+  kowl:
+    extends:
+      file: ${PATH_MANIFESTS}/deps/kafka-tools.yml
+      service: kowl
+    depends_on:
+      - zookeeper-1
+  localstack:
+    extends:
+      file: ${PATH_MANIFESTS}/deps/localstack.yml
+      service: localstack
```
Comment (lindenmckenzie) [inline v2/stacks/auth/core-ons.yml:102]: Do we need localstack for publishing?

File: v2/manifests/core-ons/the-train.yml [hunk 6]
```diff
@@ -19,7 +19,6 @@ services:
       PORT:                        8084
       DP_COLOURED_LOGGING:         "true"
       DP_LOGGING_FORMAT:           "pretty_json"
-      PUBLISHING_THREAD_POOL_SIZE: 100
```
Comment (lindenmckenzie) [inline v2/manifests/core-ons/the-train.yml:22]: Why are you removing this?

**Review by lindenmckenzie — COMMENTED** [review]
Some comments / discussion points.

File: v2/stacks/auth/.env (hunk 1 above)
Comment (gedge) [reply v2/stacks/auth/.env:17]: You're right, but alas, the files-api, image-api and download-service (and perhaps others) all use https://pkg.go.dev/strconv#ParseBool for this var

We will have to make an exception for babbage.

File: v2/stacks/auth/core-ons.yml (hunk 5 above)
Comment (gedge) [reply v2/stacks/auth/core-ons.yml:102]: it was needed for s3, which is required by the files-api, without which the publish step fails (at the last step, so sorta succeeds)

File: v2/stacks/auth/.env (hunk 1 above)
Comment (lindenmckenzie) [reply v2/stacks/auth/.env:17]: ah very good, ok then.

File: v2/stacks/auth/core-ons.yml (hunk 5 above)
Comment (lindenmckenzie) [reply v2/stacks/auth/core-ons.yml:102]: Thanks for clarification.

File: v2/stacks/auth/core-ons.yml (hunk 2 above)
Comment (dionisislolas) [reply v2/stacks/auth/core-ons.yml]: This is now addressed.

File: v2/stacks/auth/core-ons.yml (hunk 3 above)
Comment (dionisislolas) [reply v2/stacks/auth/core-ons.yml]: This is now addressed.

File: v2/stacks/auth/core-ons.yml (hunk 4 above)
Comment (dionisislolas) [reply v2/stacks/auth/core-ons.yml]: This is now addressed.

File: v2/manifests/core-ons/the-train.yml (hunk 6 above)
Comment (dionisislolas) [reply v2/manifests/core-ons/the-train.yml:22]: This was in added tow times so one was redundant.

**Pull request merged**